src/hooks/useUpdateEffect.ts

```diff
@@ -0,0 +1,15 @@
+import { useEffect, useRef } from "react";
+
+export const useUpdateEffect: typeof useEffect = (effect, deps) => {
```

Comment:
@vineethasok there is no custom hooks in click-ui as of now, so this is a vital point.
I find both custom hooks, and `useUpdateEffect` in particular, to be quite useful in certain cases. WDYT? Are you ok with adding a new module with hooks?

I can find other ways to deal with the bug, but in my mind running this effect on updates only is exactly the way to fix it.